File: backend/src/main/java/es/udc/paproject/backend/model/services/ProductService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Category;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.entities.Product;

import java.math.BigDecimal;
import java.util.List;

public interface ProductService {

    Product announceProduct(Long userId, String name, String description, long minutes, BigDecimal startPrice,
                                   String shipInfo, Long categoryId) throws InstanceNotFoundException;

    Product getProductInfo(Long productId) throws InstanceNotFoundException;

    Block<Product> findProducts(Long categoryId, String keywords, int page, int size);

    List<Category> findAllCategories();

    Block<Product> findUserProducts(Long userId, int page, int size);
}

File: frontend/src/backend/productService.js
import {config, appFetch} from './appFetch';

export const announceProduct =(name, description, minutes, startPrice, shipInfo, categoryId,
                               onSuccess, onErrors) =>
    appFetch(`/catalog/add`,
        config('POST', {name, description, minutes, startPrice, shipInfo, categoryId}),
            onSuccess, onErrors);

export const findUserProducts = ({page},
                                 onSuccess) => {
    let path = `/catalog/products/user?page=${page}`;
        appFetch(path, config('GET'), onSuccess);
}

export const findAllCategories = (onSuccess) =>
    appFetch('/catalog/categories', config('GET'), onSuccess);

export const findProducts = ({categoryId, keywords, page},
                             onSuccess) => {

    let path = `/catalog/products?page=${page}`;

    path += categoryId ? `&categoryId=${categoryId}` : "";
    path += keywords.length > 0 ? `&keywords=${encodeURIComponent(keywords)}` : "";

    appFetch(path, config('GET'), onSuccess);

}

export const findByProductId = (id, onSuccess) =>
    appFetch(`/catalog/products/${id}`, config('GET'), onSuccess);
